preview-maker.js

```diff
@@ -1,184 +1,391 @@
+      colorsHue.push(colornow);
+    }
+  }
+  if (colorsHue.length > 1) {
+    var minDifference = 360;
+    for (var colorHue of colorsHue) {
+      for (var hue of colorsHue) {
+        if (hue != colorHue) {
+          hue = Number(hue);
+          colorHue = Number(colorHue);
+          if (Math.abs(hue - colorHue) < minDifference) {
+            minDifference = Math.abs(hue - colorHue);
+            accentHue = (hue + colorHue) / 2;
+          }
         }
+      }
+    }
+  }
+  return accentHue;
+}
+const fill = (node, color) => {
+  if (node.tagName === `stop`) {
+    node.setAttribute(`stop-color`, color);
+  } else {
+    node.setAttribute(`fill`, color);
+  }
+
+  if (node.childNodes) {
+    for (let child of Array.from(node.childNodes)) {
+      if (child.setAttribute) {
+        fill(child, color);
+      }
     }
+  }
 };
-
+const preFill = (node, color) => {
+  if (color.hue == undefined) {
+    fill(
+      node,
+      `rgba(${color.red}, ${color.green}, ${color.blue}, ${color.alpha / 255})`
+    );
+  } else {
+    fill(node, `hsl(${color.hue}, ${color.saturation}%, ${color.lightness}%)`);
+  }
+};
+function rgbDifferent(color1, color2) {
+  const result = Math.hypot(
+    color1.red - color2.red,
+    color1.green - color2.green,
+    color1.blue - color2.blue
+  );
+  return result;
+}
 const makePrev = async (themeBuffer, themeName, themeAuthor, template) => {
-    let theme;
-
-    if (themeBuffer instanceof Buffer) {
-        theme = new Attheme(themeBuffer.toString(`binary`));
+  let theme, accentHue;
+
+  if (themeBuffer instanceof Buffer) {
+    theme = new Attheme(themeBuffer.toString(`binary`));
+  } else {
+    theme = themeBuffer;
+  }
+
+  const preview = parser.parseFromString(templates[template]);
+  const avatars = [
+    "avatar_backgroundRed",
+    "avatar_backgroundOrange",
+    "avatar_backgroundViolet",
+    "avatar_backgroundGreen",
+    "avatar_backgroundCyan",
+    "avatar_backgroundBlue",
+  ];
+  const inBubble =
+    theme[`chat_inBubble`] || defaultVariablesValues[`chat_inBubble`];
+  const outBubble =
+    theme[`chat_outBubble`] || defaultVariablesValues[`chat_outBubble`];
+  if (Color.brightness(inBubble) > Color.brightness(outBubble)) {
+    theme["chat_{in/out}Bubble__darkest"] = inBubble;
+  } else {
+    theme["chat_{in/out}Bubble__darkest"] = outBubble;
+  }
+  let colors = [];
+  for (const variable in defaultVariablesValues) {
+    if (variable === `chat_wallpaper` && !theme.chat_wallpaper) {
+      continue;
+    }
+    const elements = getElementsByClassName(preview, variable);
+    const color = theme[variable] || defaultVariablesValues[variable];
+    for (const element of elements) {
+      preFill(element, color);
+    }
+    const windowBackgroundWhite = rgbToHsl(
+      theme["windowBackgroundWhite"] ||
+        defaultVariablesValues["windowBackgroundWhite"]
+    );
+    const chooseHsl = rgbToHsl(color);
+    if (
+      Math.abs(chooseHsl.hue - windowBackgroundWhite.hue) > 6 &&
```

Comment:
Suppose that `chooseHsl.hue` is `359` and `windowBackgroundWhite.hue` is `1`. The distance between such colors is 2°, but this simple subtraction will return 358° and therefore two similar colors may be pushed to `colors`.